internal/command/action_test.go

```diff
@@ -85,6 +86,42 @@ func TestJoinEnv(t *testing.T) {
 	})
 }
 
+func TestDefaultVariableResolution(t *testing.T) {
+	Convey("Variable resolution correctly resolves variables", t, func() {
+		expectedDefaultValue := "defaultValueOfVariable"
+
+		dir, err := ioutil.TempDir("", "summon")
+		So(err, ShouldBeNil)
+		if err != nil {
+			return
+		}
+		defer os.RemoveAll(dir)
+
+		tempFile := filepath.Join(dir, "outputFile.txt")
+
+		err = runAction(&ActionConfig{
+			Args:       []string{"bash", "-c", "echo -n \"$FOO\" > " + tempFile},
+			YamlInline: "FOO: !str:default='" + expectedDefaultValue + "'",
```

Comment:
IIUC, this test case covers when the FOO env is not set, but a default is set. For completeness, do we need a test case for:
* Environment (or Provider's value) is set AND default is set (default gets overrided)

Reply:
Yeah it was a bit odd since the old code had no tests at all. I'll add the test case you specified.

Reply:
Done